### PR TITLE
testing/clazy: new aport

### PR DESCRIPTION
**Pull request by misery**

### CODE REVIEW

**Comment by misery** [comment]
work in progress......


https://bugs.kde.org/show_bug.cgi?id=379248

**Comment by kaniini** [comment]
this bug is actually a defect in the clang/llvm static libraries, we need to debug it.

**Comment by misery** [comment]
Still broken with clang 5 :-(

```
alpine:~# clazy main.cpp 
: CommandLine Error: Option 'help-list' registered more than once!
fatal error: error in backend: inconsistency in registered CommandLine options
clang-5.0: error: clang frontend command failed with exit code 70 (use -v to see invocation)
Alpine clang version 5.0.0 (tags/RELEASE_500/final) (based on LLVM 5.0.0)
Target: x86_64-alpine-linux-musl
Thread model: posix
InstalledDir: /usr/bin
clang-5.0: note: diagnostic msg: PLEASE submit a bug report to  and include the crash backtrace, preprocessed source, and associated run script.
clang-5.0: note: diagnostic msg: Error generating preprocessed source(s).
```

**Comment by misery** [comment]
Any idea?

**Comment by clandmeter** [comment]
@kaniini any update on this or should we close it?

**Comment by misery** [comment]
Well, still want to see this merged if the bug is closed.

**Comment by misery** [comment]
Any hint?
```
Scanning dependencies of target clazy-standalone
[ 99%] Building CXX object CMakeFiles/clazy-standalone.dir/src/ClazyStandaloneMain.cpp.o
[100%] Linking CXX executable bin/clazy-standalone
/usr/bin/ld: CMakeFiles/clazy-standalone.dir/src/ClazyStandaloneMain.cpp.o: in function `main':
ClazyStandaloneMain.cpp:(.text+0x45): undefined reference to `clang::tooling::CommonOptionsParser::CommonOptionsParser(int&, char const**, llvm::cl::OptionCategory&, llvm::cl::NumOccurrencesFlag, char const*)'
/usr/bin/ld: ClazyStandaloneMain.cpp:(.text+0x91): undefined reference to `clang::PCHContainerOperations::PCHContainerOperations()'
/usr/bin/ld: ClazyStandaloneMain.cpp:(.text+0xa6): undefined reference to `clang::vfs::getRealFileSystem()'
/usr/bin/ld: ClazyStandaloneMain.cpp:(.text+0xc1): undefined reference to `clang::tooling::ClangTool::ClangTool(clang::tooling::CompilationDatabase const&, llvm::ArrayRef<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::shared_ptr<clang::PCHContainerOperations>, llvm::IntrusiveRefCntPtr<clang::vfs::FileSystem>)'
/usr/bin/ld: ClazyStandaloneMain.cpp:(.text+0x132): undefined reference to `clang::tooling::ClangTool::run(clang::tooling::ToolAction*)'
/usr/bin/ld: ClazyStandaloneMain.cpp:(.text+0x13d): undefined reference to `clang::tooling::ClangTool::~ClangTool()'
/usr/bin/ld: CMakeFiles/clazy-standalone.dir/src/ClazyStandaloneMain.cpp.o: in function `ClazyToolActionFactory::~ClazyToolActionFactory()':
ClazyStandaloneMain.cpp:(.text._ZN22ClazyToolActionFactoryD0Ev[_ZN22ClazyToolActionFactoryD0Ev]+0x5): undefined reference to `clang::tooling::FrontendActionFactory::~FrontendActionFactory()'
/usr/bin/ld: CMakeFiles/clazy-standalone.dir/src/ClazyStandaloneMain.cpp.o: in function `_GLOBAL__sub_I_ClazyStandaloneMain.cpp':
ClazyStandaloneMain.cpp:(.text.startup+0xbeb): undefined reference to `clang::tooling::CommonOptionsParser::HelpMessage'
/usr/bin/ld: CMakeFiles/clazy-standalone.dir/src/ClazyStandaloneMain.cpp.o:(.data.rel.ro._ZTV22ClazyToolActionFactory[_ZTV22ClazyToolActionFactory]+0x10): undefined reference to `clang::tooling::FrontendActionFactory::~FrontendActionFactory()'
/usr/bin/ld: CMakeFiles/clazy-standalone.dir/src/ClazyStandaloneMain.cpp.o:(.data.rel.ro._ZTV22ClazyToolActionFactory[_ZTV22ClazyToolActionFactory]+0x20): undefined reference to `clang::tooling::FrontendActionFactory::runInvocation(std::shared_ptr<clang::CompilerInvocation>, clang::FileManager*, std::shared_ptr<clang::PCHContainerOperations>, clang::DiagnosticConsumer*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclContext::getRedeclContext()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::MatchFinder::MatchCallback::~MatchCallback()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTConsumer::HandleTopLevelDecl(clang::DeclGroupRef)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXRecordDecl::DefinitionData::getBasesSlowCase() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_type_traits::ASTNodeKind::getFromNode(clang::Decl const&)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getTemplateSpecializationInfo() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FrontendAction::~FrontendAction()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FriendDecl::getNextFriendSlowCase()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TypedefType const* clang::Type::getAs<clang::TypedefType>() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::LambdaExpr::isInitCapture(clang::LambdaCapture const*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::getRawToken(clang::SourceLocation, clang::Token&, clang::SourceManager const&, clang::LangOptions const&, bool)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTContext::getPointerType(clang::QualType) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::VarDecl::getInit()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::LambdaExpr::getCallOperator() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::internal::DynTypedMatcher::tryBind(llvm::StringRef) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getFileCharacteristic(clang::SourceLocation) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Expr::getBestDynamicClassTypeExpr() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getBody(clang::FunctionDecl const*&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getImmediateExpansionRange(clang::SourceLocation) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Decl::castFromDeclContext(clang::DeclContext const*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DiagnosticIDs::getCustomDiagID(clang::DiagnosticIDs::Level, llvm::StringRef)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::QualType::getSplitUnqualifiedTypeImpl(clang::QualType)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::Lex(clang::Token&)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMemberCallExpr::getImplicitObjectArgument() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTFrontendAction::ExecuteAction()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::getImmediateMacroName(clang::SourceLocation, clang::SourceManager const&, clang::LangOptions const&)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CallExpr::getDirectCallee()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclContext::using_directives() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::StmtIteratorBase::NextDecl(bool)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::QualType::getAsString[abi:cxx11]() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CapturedDecl::getBody() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::PPCallbacks::~PPCallbacks()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXRecordDecl::getDestructor() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMethodDecl::isStatic() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getNumParams() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::getAsTagDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMethodDecl::size_overridden_methods() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Expr::getBestDynamicClassType() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParmVarDecl::getDefaultArg()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getSpellingLineNumber(clang::SourceLocation, bool*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::getLocForEndOfToken(clang::SourceLocation, unsigned int, clang::SourceManager const&, clang::LangOptions const&)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMemberCallExpr::getRecordDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXForRangeStmt::getRangeInit()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::LambdaExpr::captures() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::StmtIteratorBase::GetDeclExpr() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXConstructExpr::getBeginLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::internal::DynTypedMatcher::dynCastTo(clang::ast_type_traits::ASTNodeKind) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CallExpr::getCalleeDecl()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::isClassType() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParentMap::addStmt(clang::Stmt*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXForRangeStmt::getLoopVariable()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FrontendAction::shouldEraseOutputFiles()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DiagnosticsEngine::EmitCurrentDiagnostic(bool)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_type_traits::ASTNodeKind::getFromNode(clang::Stmt const&)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TemplateName::getAsQualifiedTemplateName() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXRecordDecl::getFirstFriend() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::NamespaceDecl::getOriginalNamespace()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FixItRewriter::WriteFixedFiles(std::vector<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXConstructorDecl::init_begin() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getTemplateSpecializationArgs() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::LambdaExpr::capture_begin() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FrontendAction::FrontendAction()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::isIncompleteType(clang::NamedDecl**) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `vtable for clang::Lexer'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclContext::decls_begin() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TagType::getDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::getUnqualifiedDesugaredType() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParentMap::getParent(clang::Stmt*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::MatchFinder::MatchCallback::getID() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTConsumer::HandleInterestingDecl(clang::DeclGroupRef)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::getSourceText(clang::CharSourceRange, clang::SourceManager const&, clang::LangOptions const&, bool*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::MatchFinder::addMatcher(clang::ast_matchers::internal::Matcher<clang::Stmt> const&, clang::ast_matchers::MatchFinder::MatchCallback*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::isTemplateInstantiation() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParmVarDecl::hasDefaultArg() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getLanguageLinkage() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclContext::udir_iterator::operator*() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getDecomposedExpansionLocSlowCase(clang::SrcMgr::SLocEntry const*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclarationName::getNameKind() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::MemberPointerType::getMostRecentCXXRecordDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::NamedDecl::getLinkageInternal() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclContext::getDeclKindName() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TemplateName::TemplateName(void*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXRecordDecl::getTemplateSpecializationKind() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getPresumedLoc(clang::SourceLocation, bool) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Decl::castToDeclContext(clang::Decl const*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::getAsCXXRecordDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TemplateName::getAsDependentTemplateName() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTConsumer::HandleTopLevelDeclInObjCContainer(clang::DeclGroupRef)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Stmt::getSourceRange() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Stmt::dump() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getBufferData(clang::FileID, bool*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Stmt::getEndLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclaratorDecl::getOuterLocStart() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CapturedStmt::getCapturedDecl()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParmVarDecl::getUninstantiatedDefaultArg()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getSpellingLocSlowCase(clang::SourceLocation) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::QualType::getAsString[abi:cxx11](clang::PrintingPolicy const&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getOverloadedOperator() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Decl::isInAnonymousNamespace() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Stmt::children()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::PluginASTAction::anchor()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::Lexer(clang::SourceLocation, clang::LangOptions const&, char const*, char const*, char const*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Expr::IgnoreCasts()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::getSpelling[abi:cxx11](clang::Token const&, clang::SourceManager const&, clang::LangOptions const&, bool*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `llvm::Registry<clang::PluginASTAction>::add_node(llvm::Registry<clang::PluginASTAction>::node*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getTemplateInstantiationPattern() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getFileLocSlowCase(clang::SourceLocation) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CallExpr::getEndLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getFileIDSlow(unsigned int) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Lexer::Lexer(clang::FileID, llvm::MemoryBuffer const*, clang::SourceManager const&, clang::LangOptions const&)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ObjCInterfaceDecl::LoadExternalDefinition() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TemplateName::TemplateName(clang::TemplateDecl*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::NamedDecl::getQualifiedNameAsString[abi:cxx11]() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParentMap::ParentMap(clang::Stmt*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::loadSLocEntry(unsigned int, bool*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SubstTemplateTypeParmPackType::getArgumentPack() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMemberCallExpr::getMethodDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTConsumer::HandleImplicitImportDecl(clang::ImportDecl*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXRecordDecl::getDescribedClassTemplate() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMethodDecl::isCopyAssignmentOperator() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXConstructorDecl::isCopyOrMoveConstructor(unsigned int&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Stmt::getBeginLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::NestedNameSpecifierLoc::getTypeLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getCharacterData(clang::SourceLocation, bool*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Expr::IgnoreParenImpCasts()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceLocation::printToString[abi:cxx11](clang::SourceManager const&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::isDefined(clang::FunctionDecl const*&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::getUnusedResultAttr() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::StmtIteratorBase::NextVA()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FullSourceLoc::isBeforeInTranslationUnitThan(clang::SourceLocation) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Decl::getAttrs() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::LambdaCapture::getCaptureKind() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXConstructorDecl::isDefaultConstructor() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `vtable for clang::PPChainedCallbacks'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::internal::DynTypedMatcher::constructVariadic(clang::ast_matchers::internal::DynTypedMatcher::VariadicOperator, clang::ast_type_traits::ASTNodeKind, std::vector<clang::ast_matchers::internal::DynTypedMatcher, std::allocator<clang::ast_matchers::internal::DynTypedMatcher> >)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Decl::hasDefiningAttr() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::UsingDirectiveDecl::getNominatedNamespace()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclarationName::getAsString[abi:cxx11]() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::internal::DynTypedMatcher::matches(clang::ast_type_traits::DynTypedNode const&, clang::ast_matchers::internal::ASTMatchFinder*, clang::ast_matchers::internal::BoundNodesTreeBuilder*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FunctionDecl::isVariadic() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::VarDecl::getTemplateSpecializationKind() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CallExpr::getBeginLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::getPointeeType() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclarationName::getCXXDeductionGuideTemplate() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SrcMgr::ContentCache::getBuffer(clang::DiagnosticsEngine&, clang::SourceManager const&, clang::SourceLocation, bool*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMethodDecl::hasInlineBody() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::QualType::isTriviallyCopyableType(clang::ASTContext const&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_type_traits::ASTNodeKind::isBaseOf(clang::ast_type_traits::ASTNodeKind, unsigned int*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Stmt::getStmtClassName() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::charinfo::InfoTable'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXConstructorDecl::isCopyConstructor(unsigned int&) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::getPointeeCXXRecordDecl() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `vtable for clang::PreprocessorLexer'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::MacroDirective::getDefinition()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ASTContext::getTypeInfo(clang::Type const*) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParentMap::setParent(clang::Stmt const*, clang::Stmt const*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FixItRewriter::FixItRewriter(clang::DiagnosticsEngine&, clang::SourceManager&, clang::LangOptions const&, clang::FixItOptions*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::internal::DynTypedMatcher::trueMatcher(clang::ast_type_traits::ASTNodeKind)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ast_matchers::internal::HasNameMatcher::HasNameMatcher(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ReturnStmt::getRetValue()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::ParentMap::~ParentMap()'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getFakeBufferForRecovery() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::TypeLoc::getLocalAlignmentForType(clang::QualType)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::LambdaExpr::getBody() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::isFloatingType() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::MemberExpr::getEndLoc() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::NestedNameSpecifier::getKind() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::DeclContext::classof(clang::Decl const*)'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::CXXMethodDecl::overridden_methods() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::SourceManager::getExpansionLocSlowCase(clang::SourceLocation) const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::RecordDecl::field_begin() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::Type::isCharType() const'
/usr/bin/ld: lib/ClazyPlugin.so: undefined reference to `clang::FixItOptions::~FixItOptions()'
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

File: testing/clazy/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Maintainer: André Klitzing <aklitzing@gmail.com>
+pkgname=clazy
+pkgver=1.5
+pkgrel=0
+pkgdesc="Qt oriented code checker based on clang framework"
+url="https://github.com/KDE/clazy"
+arch="all"
+license="GPLv2"
```
Comment (tcely) [inline testing/clazy/APKBUILD]: https://spdx.org/licenses/

Please choose the correct identifier from above list.

File: testing/clazy/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+makedepends="llvm-dev clang-dev cmake"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.gz::https://github.com/KDE/$pkgname/archive/v$pkgver.tar.gz"
+builddir="$srcdir/"$pkgname-$pkgver
```
Comment (tcely) [inline testing/clazy/APKBUILD]: Specifying the default `builddir` here is optional.

File: testing/clazy/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+build() {
+	cd "$builddir"
```
Comment (tcely) [inline testing/clazy/APKBUILD]: The current version of `abuild` does this `cd` before running the function.

File: testing/clazy/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	cd "$builddir/build"
+	chmod 755 clazy
+	chmod 755 clazy-standalone
+	./clazy --version
+	./clazy-standalone --version
+}
+
+package() {
+	cd "$builddir/build"
+	make DESTDIR="$pkgdir" install
```
Comment (tcely) [inline testing/clazy/APKBUILD]:
```suggestion
	make -C build DESTDIR="$pkgdir" install
```

File: testing/clazy/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+check() {
+	cd "$builddir/build"
+	chmod 755 clazy
+	chmod 755 clazy-standalone
+	./clazy --version
+	./clazy-standalone --version
+}
+
+package() {
+	cd "$builddir/build"
```
Comment (tcely) [inline testing/clazy/APKBUILD]:
```suggestion
```

**Comment by Cogitri** [comment]
Stale

**Pull request closed without merging**